docs/okeps/okep-5259-no-overlay.md

```diff
@@ -0,0 +1,595 @@
+but I don't want to deploy an external BGP router for the OCP cluster.
+
+## Proposed Solution
+
+The core idea is to leverage the existing Route Advertisement feature to
+advertise the IP subnet allocated to each node (which contains the IPs of pods
+running on that node) to an internal/external BGP route reflector. The BGP route
+reflector will then populate these routes throughout the physical network,
+allowing other nodes to directly route traffic to pod IPs without needing an
+overlay. Within a cluster, networks can operate in different overlay modes.
+Users can have overlay and no-overlay networks simultaneously.
+
+When the no-overlay mode is enabled for a network:
+
+* For north-south traffic, we will follow the implementation of the Route
+  Advertisement feature, the traffic from pods will no longer be SNATed on
+  egress. In the route advertisement feature, SNAT is disabled for all pod
+  egress traffic that leaves the node, regardless of whether the destination
+  route is BGP learnt or not. For users who only want to have no-overlay mode
+  but not want to expose podIPs to external, this behavior is not ideal. Instead
+  of disabling SNAT for all the pod-to-external traffic, OVN-K shall adjust its
+  SNAT behavior so that pod IPs are exposed for traffic routed via BGP, but SNAT
+  is applied for non-BGP-routed egress traffic.
+
+* For east-west traffic, intra-node traffic (pod-to-pod, pod-to-clusterIP and
+  pod-to-host) remains unchanged, while cross-node traffic will follow the same
+  path as north-south traffic.
+
+### API Details
+
+We introduce a new `layer3.encapsulation` field to be added to the Spec of the
+ClusterUserDefinedNetwork (CUDN) CRD. This
+new field will enable control over the network transport behavior with the
+following options:
+
+1. **Encapsulation Configuration**:
+   * `encapsulation`: specifies the transport method (default: `Geneve`)
+   * Supported values: `Geneve` (default) or `None` (no overlay)
+
+The `spec` field of a ClusterUserDefinedNetwork CR is immutable. Therefore, the
+encapsulation configuration cannot be changed once a ClusterUserDefinedNetwork
+CR is created.
+
+#### Example of a layer-3 CUDN with no-overlay mode enabled
+
+A layer-3 CUDN that enables no-overlay mode should look like this:
+
+```yaml
+apiVersion: k8s.ovn.org/v1
+kind: ClusterUserDefinedNetwork
+metadata:
+  name: my-cudn
+spec:
+  namespaceSelector:
+    matchExpressions:
+    - key: kubernetes.io/metadata.name
+      operator: In
+      values: ["red", "blue"]
+  network:
+    topology: Layer3
+    layer3:
+      role: Primary
+      mtu: 1500
+      subnets:
+      - cidr: 10.10.0.0/16
+        hostSubnet: 24
+      encapsulation: None
+```
+
+#### Example of the default network with no-overlay mode enabled
+
+For the cluster default network, the cluster admin shall create a
+ClusterUserDefinedNetwork CR with the reserved name `default` in the namespace where
```

Comment:
is that possible enable no-overlay node for default network by CNO like `no-overlay-default-network: enable` and let CNO to create a default ClusterUserDefinedNetwork CR ?